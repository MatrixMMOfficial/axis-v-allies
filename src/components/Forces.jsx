import React from 'react'
import { UnitImg } from './UnitFigure'

const Forces = ({ units, classNameFct, handleClick }) => (
  units.map(({ id, power, type }) => (
    <UnitImg 
      key={id} 
      power={power} 
      className={classNameFct(id)}
      handleClick={handleClick(id, type)}
      name={type} />
  ))
)

Forces.defaultProps = {
  classNameFct: () => null,
  handleClick: () => null
}
export default Forces
