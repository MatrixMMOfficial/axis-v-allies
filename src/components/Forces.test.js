'use strict'

import React from 'react'
import Forces from './Forces'
import renderer from 'react-test-renderer'


it('correctly renders a pair of German infantry', () => {
  const units = [
    { id: 1, power: 'Germany', type: 'infantry' },
    { id: 2, power: 'Germany', type: 'infantry' }
  ]
  const tree = renderer.create(
    <Forces units={units} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

