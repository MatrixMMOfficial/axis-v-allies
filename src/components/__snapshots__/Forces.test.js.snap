// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly renders a pair of German infantry 1`] = `
Array [
  <img
    alt="Germany infantry"
    className={null}
    onClick={null}
    src="infantry.png"
/>,
  <img
    alt="Germany infantry"
    className={null}
    onClick={null}
    src="infantry.png"
/>,
]
`;
